compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -582,6 +583,19 @@ fn try_prove_negated_where_clause<'tcx>(
         return false;
     };
 
+    // PERF(sized-hierarchy): Sizedness supertraits aren't elaborated to improve perf, so
+    // check for a `Sized` subtrait when looking for `MetaSized`. `PointeeSized` bounds
+    // are syntactic sugar for a lack of bounds so don't need this.
+    if let Some(pred) = clause.as_trait_clause()
+        && is_unelaborated_sizedness_optimisation(
+            root_infcx,
+            pred,
+            param_env.caller_bounds().iter().filter_map(|c| c.as_trait_clause()),
+        )
+    {
+        return false;
+    }
```

Comment:
why is that necessary. I would expect the change in the trait solver itself to be sufficient.

Reply:
I thought this was necessary when I added it but I presume the changes in the trait solver were broken at that point and I mistook that for this being necessary, removed now

Reply:
I thought this was necessary as part of the elaboration optimisation, but in attempting to re-add something like https://github.com/rust-lang/rust/pull/137944#discussion_r2059089523, it turns out it's that change which made this necessary.